tests/test_interrupted_sync.py

```diff
@@ -62,26 +62,45 @@ def get_bookmark_value(self, state, stream):
         return None
 
     ##########################################################################
-    # Tests To Override
+    # Tests and Functions To Override
     ##########################################################################
 
-    def test_bookmarked_streams_start_date(self):
-        # TODO - BUG - completed streams are not respecting the bookmark value
-        def override_lookback_window(stream=None):
-            look_back = {stream: timedelta(days=0) for stream in self.streams_to_test()}
-            if stream:
-                return look_back[stream]
-            return look_back
-
-        self.expected_lookback_window = override_lookback_window
-        super().test_bookmarked_streams_start_date()
-
-    def test_resuming_sync_records(self):
-        # TODO - BUG - completed streams are not respecting the bookmark value
-        def override_lookback_window(stream=None):
-            look_back = {stream: timedelta(days=0) for stream in self.streams_to_test()}
-            if stream:
-                return look_back[stream]
-            return look_back
-        self.expected_lookback_window = override_lookback_window
-        super().test_resuming_sync_records()
+    def calculate_expected_sync_start_time(self, bookmark, stream, completed=True):
+        """This method is only for streams that have bookmarks and a sync has been started"""
+
+        # BUG override bookmark to use final state vs manipulate_state allowing test to pass
+        done_status_list = [ "Closed", "Done", "Rejected" ]
+        if not GA4InterruptedSyncTest.jira_status:
+            GA4InterruptedSyncTest.jira_status = get_jira_card_status('TDL-23687')
+        self.assertNotIn(GA4InterruptedSyncTest.jira_status, done_status_list,
+                         msg="JIRA BUG has transitioned to Done, remove work around")
+        bookmark  = self.get_bookmark_value(self.resuming_sync_state, stream)
+
+        # The lookback window should be used for completed streams
+        lookback = self.expected_lookback_window()
+        if type(lookback) is dict:
+            # lookback was retrieved from base_case or override all streams (dict)
+            stream_lookback = lookback.get(stream)
+        elif type(lookback) is datetime.timedelta:
+            # lookback was retrieved from base_case or override single stream (timedelta)
+            stream_lookback = lookback
+        else:
+            raise TypeError
```

Comment:
The `tap_tester.BaseCase` function returns a dictionary. It feels like a design mistake if we make the overrides of `expected_lookback_window` return anything but a dictionary

Reply:
I read this as being able to return the whole dict or just one of the keys if a stream is provided in the signature so I wanted to be able to handle both.  Please advise:

https://github.com/stitchdata/tap-tester/blob/master/tap_tester/base_suite_tests/base_case.py#L204C1-L215C1